js/components/viewControlsVr.js

```diff
@@ -21,35 +22,34 @@ var ViewControlsVr = React.createClass({
   },
 
   componentWillMount: function () {
+    // if we have vr mode, and the device !== mobile, we need to add control element to the screen of the player.
+    // control element is covered with icon from fonts
     this.isMobile = false;
     this.vr = null;
     this.icon = {};
-
     if (this.props.controller) {
       if (this.props.controller.videoVrSource) {
         this.vr = this.props.controller.videoVrSource.vr;
       }
-
       if (this.props.controller.state) {
         this.isMobile = this.props.controller.state.isMobile;
       }
-
-
-      if (!this.props.controller.state.isPlayingAd) {
-        if (!this.isMobile) {
-          var desktopContent = this.props.skinConfig.buttons.desktopContent;
-          this.icon = _.find(desktopContent, function (el) {
-            return el.location === "mainView";
-          });
+      if (!(this.props.controller.state.isPlayingAd || this.isMobile)) {
+        if (!(this.props.skinConfig && this.props.skinConfig.buttons && _.isArray(this.props.skinConfig.buttons.desktopContent))) {
+          return;
         }
+        var desktopContent = this.props.skinConfig.buttons.desktopContent;
+        this.icon = _.find(desktopContent, function (el) {
```

Comment:
I think that this project uses underscore as a dev dependency only, so including it might be increasing the file size somewhat. It's probably not a big deal, but the plugin receives an underscore reference during initialization, so there might be a way to reference that instead of requiring it. What do you guys think @pilievOoyala @aeng7 ?

Reply:
True, the less we need extra library, the better. We should utilize the provided underscore from plugin.

Reply:
Connecting the library underscore is necessary for the passage of unit-tests. When executing the code, underscore ("_") is in the global scope, but when it runs tests it is not there.
We can write our solution for find in an array, but I think that this is not the best solution

Reply:
Yes, we've been using it for unit tests, that part is fine. For the actual code it might be worthwhile to see if we can use the underscore reference from the global scope.